packages/lib-sourcify/src/lib/verification.ts

```diff
@@ -59,7 +59,26 @@ export async function verifyDeployed(
     deployedBytecode,
     recompiled.immutableReferences
   );
-  if (match.status) return match;
+  if (match.status === 'perfect') return match;
+
+  if (
+    match.status === 'partial' &&
+    (await checkedContract.tryToFindOriginalMetadata(deployedBytecode))
+  ) {
+    const originalMetadataMatch = { ...match };
+    const recompiled = await checkedContract.recompile();
+
+    matchWithDeployedBytecode(
+      originalMetadataMatch,
+      recompiled.deployedBytecode,
+      deployedBytecode
+    );
+    if (originalMetadataMatch.status === 'perfect') {
+      return originalMetadataMatch;
+    } else {
+      return match;
```

Comment:
Here we return the previous `match` but we already modified the `checkedContract` in the `tryToFind...`, so the wrong checkedContract will be saved in the repo. If it's not a perfect match, we should not modify the `checkedContract` at all. 

My suggestion would be to return `new CheckedContract` or `null` from the `tryToFind..`. So first check if `match.status === 'partial'`, then `const newCheckedContract = checkedContract.tryToFind`, recompile, try to match, and if perfect `checkedContract = newCheckedContract`

Reply:
I think here you misanderstood, or I'm not seeing something. In theory the `initSolcJsonInput` method is called only if we have a perfect match, so this.solidity, this.metadata, this.metadataRaw are not updated by `tryToFindOriginalMetadata`.